nix/lib/default.nix

```
@@ -1,8 +1,28 @@
-lib: {
-  dashToUnderscore =
-    attrset:
-    lib.mapAttrs' (name: value: {
-      name = lib.replaceStrings [ "-" ] [ "_" ] name;
-      inherit value;
-    }) attrset;
+lib: rec {
+  kebabToSnakeCase = lib.replaceStrings [ "-" ] [ "_" ];
+
+  kebabToSnakeCaseAttrs = lib.mapAttrs' (
+    name: value: lib.nameValuePair (kebabToSnakeCase name) value
+  );
+
+  dashToUnderscoreRecursive =
+    attrs:
+    let
+      transformAttr =
+        attr:
+        let
+          value = attrs.${attr};
+        in
+        {
+          name = lib.replaceStrings [ "-" ] [ "_" ] attr;
```

```suggestion
          name = dashToUnderscore attr;
```

We can reuse the existing function, no need to write it out twice.